homeassistant/components/aws/image_processing.py

```diff
@@ -0,0 +1,483 @@
+        self.identify_faces = entity_config.get(CONF_IDENTIFY_FACES, False)
+        self.conf_save_file_folder = entity_config.get(CONF_SAVE_FILE_FOLDER, None)
+        self.conf_save_file_timestamp = entity_config.get(
+            CONF_SAVE_FILE_TIMESTAMP, False
+        )
+        self._confidence = entity_config.get(CONF_CONFIDENCE, 0.0)
+        self.detection_attributes = entity_config.get(
+            CONF_DETECTION_ATTRIBUTES, "DEFAULT"
+        )
+        self.valid_collection_id = False
+        if name:
+            self._name = name
+        else:
+            self._name = f"Rekognition Face {split_entity_id(camera_entity)[1]}"
+
+    @property
+    def confidence(self):
+        """Return minimum confidence for send events."""
+        return self._confidence
+
+    @property
+    def camera_entity(self):
+        """Return camera entity id from process pictures."""
+        return self.camera
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    async def init_collection(self, client):
+        """Create AWS image collection if doesn't exist."""
+        collection_id_req = await client.list_collections(MaxResults=99)
+        if self.collection_id not in collection_id_req["CollectionIds"]:
+            await client.create_collection(CollectionId=self.collection_id)
+        self.valid_collection_id = True
+
+    async def compute_faces(self, client, image):
+        """Detect faces in image and parse response."""
+        index_faces = await client.index_faces(
+            CollectionId=self.collection_id,
+            Image={"Bytes": image},
+            ExternalImageId=self.camera,
+            DetectionAttributes=[self.detection_attributes],
+        )
+        known_faces = []
+        detected_faces = index_faces.get("FaceRecords", [])
+        for face_record in detected_faces:
+            known_face = {}
+            box = await compute_box(face_record["Face"]["BoundingBox"], 3)
+            known_face[ATTR_NAME] = "Unknown"
+            known_face[ATTR_CONFIDENCE] = face_record["Face"]["Confidence"]
+            known_face["bounding_box"] = box["bounding_box"]
+            known_face["centroid"] = box["centroid"]
+            face_id = face_record["Face"]["FaceId"]
+            # Add extra attributes
+            if self.detection_attributes == "ALL":
+                face_detail = face_record["FaceDetail"]
+                age_range = face_detail["AgeRange"]
+                known_face[ATTR_GENDER] = face_detail["Gender"]["Value"]
+                known_face[ATTR_MOTION] = face_detail["Emotions"][0]["Type"]
+                known_face[ATTR_GLASSES] = face_detail["Eyeglasses"]["Value"]
+                known_face[ATTR_AGE] = (
+                    float(age_range["High"]) + float(age_range["Low"])
```

Comment:
I am interested to publish the high and low values, might be useful for detecting children for example

Reply:
Hmm, feels against the spirit of the base component to return a `dict` for `ATTR_AGE` but it does make sense to keep processing to a minimum and return the actual values to the user. Will implement as can easily change if needed 😄

Reply:
Might also raise some more questions about returning the full range of attributes. Rekognition returns a wealth of information which get parsed out in processing which isn't ideal: `Smile Sunglasses Beard Mustache EyesOpen MouthOpen`. Given the generic nature of the attributes feels like these should be new constants added in the base `image_processing` component.

Reply:
Actually I meant like: `known_face[ATTR_AGE_MIN]=float(age_range["Low"]`.
Agreed its nice to share the full returned data, and including it in event seemed a good way to go